flybywiresim/a32nx
# refactor: Improving simbridge-client connection handling

## Summary of Changes
This PR improves the simbridge-client for interaction with the SimBridge by introducing a client state to check for SimBridge availability to avoid all services having to check for themselves. 

It respects flyPad Settings-SimOptions - SimBridge Enabled AUTO/OFF

It will include the time limited connection attempts from the MCDU to avoid polling the SimBridge for ever with the unavoidable error messages in the console and performance issues we have encountered.

The handling of the MCDU client connection for the MCDU to the MCDU Server moved to the simbridge-client. 

Discord username (if different from GitHub): Cdr_Maverick#6475

## Testing instructions
- Test normal operations with features using SimBridge - SimBridge must be started
  - EFB Health Status (Status Bar on top - the Wifi-like symbol)
  - Local Files
  - MCDU Server
  - CoRoutes
  - (Terrain) - can only be tested when this is deployed to Experimental
- Shut Down SimBridge - features should no longer be available but gracefully fail
  - Check flyPad Settings-SimOptions - SimBridge Enabled should show AUTO and Active
  - Wait 5+ min - flyPad Settings-SimOptions - SimBridge Enabled should show AUTO and Inactive
- Start SimBridge 
  - Feature should NOT come online again as setting is still inactive
-  Set setting to OFF for 2+sec and back to AUTO again -> features should be back available and remote MCDU should reconnect (might need a reload the remote MCDU in the browser)
    -  Terrain - turn off Terr on ND for a few seconds and back on - it should reconnect and show terrain again
- When all is back turn  flyPad Settings-SimOptions - SimBridge Enabled to OFF -> features should now be unavailable as if the SimBridge is turned off 
  - Remote MCDU Screen should clear
- Turn the setting back to AUTO - all features should be available again, remote MCDU should reconnect 

Any other ideas to test this are welcome - try to break it!


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
